File: tests/fixtures/python/complex_method.py
"""Function with high cyclomatic complexity (cc >= 9)."""


def process_order(order):
    if order is None:
        return None
    if order.status == "pending":
        if order.payment_verified:
            if order.items_in_stock:
                order.status = "processing"
            else:
                order.status = "backordered"
        else:
            order.status = "payment_failed"
    elif order.status == "processing":
        if order.shipped:
            order.status = "shipped"
        elif order.cancelled:
            order.status = "cancelled"
    elif order.status == "shipped":
        if order.delivered:
            order.status = "delivered"

    return order

File: tests/fixtures/typescript/complex_method.ts
function processOrder(order: any): any {
    if (order === null) {
        return null;
    }
    if (order.status === "pending") {
        if (order.paymentVerified) {
            if (order.itemsInStock) {
                order.status = "processing";
            } else {
                order.status = "backordered";
            }
        } else {
            order.status = "payment_failed";
        }
    } else if (order.status === "processing") {
        if (order.shipped) {
            order.status = "shipped";
        } else if (order.cancelled) {
            order.status = "cancelled";
        }
    } else if (order.status === "shipped") {
        if (order.delivered) {
            order.status = "delivered";
        }
    }
    return order;
}
